### PR TITLE
Remove support for multiclass (buggy) in SliceCombiner

### PR DESCRIPTION
**Pull request by vincentschen**
* Raises `NotImplementedError` in multiclass setting

**Test Plan**
* Fixed existing tests 
* `tox` + `tox -e complex` passes

### CODE REVIEW

File: test/slicing/test_slice_combiner.py [hunk 1]
```diff
@@ -199,17 +199,16 @@ def test_combiner_multiclass(self):
 
         # Ensure smoother temperature for multi-class
         combiner_module = SliceCombinerModule()
-        combined_rep = combiner_module(outputs)
-        # higher tolerance because randomness in non-max class outputs -> for softmax
-        self.assertTrue(
-            torch.allclose(combined_rep, torch.ones(batch_size, h_dim) * 3, atol=1e-2)
-        )
+        with self.assertRaisesRegexp(
```
Comment (henryre) [inline test/slicing/test_slice_combiner.py]: `assertRaisesRegexp` is deprecated, use `assertRaisesRegex`

File: test/slicing/test_slice_combiner.py [hunk 2]
```diff
@@ -199,17 +199,16 @@ def test_combiner_multiclass(self):
...
+            NotImplementedError, "SliceCombiner does not support multiclass labels."
```
Comment (henryre) [inline test/slicing/test_slice_combiner.py]: nit: use a shorter chunk of error message to fit on one line

File: test/slicing/test_slice_combiner.py [hunk 3]
```diff
@@ -253,6 +252,8 @@ def test_temperature(self):
 
         # Check number of elements that match either of the original weights
         num_matching_original = torch.sum(
-            torch.isclose(combined_rep, torch.ones(batch_size, h_dim) * 2)
-        ) + torch.sum(torch.isclose(combined_rep, torch.ones(batch_size, h_dim) * 4))
+            torch.isclose(combined_rep, torch.ones(batch_size, h_dim) * 2, atol=1e-4)
+        ) + torch.sum(
```
Comment (henryre) [inline test/slicing/test_slice_combiner.py]: nit: hard to read on multiline. can we name the two torch.sum ouputs something meaningful?

File: snorkel/slicing/modules/slice_combiner.py [hunk 4]
```diff
@@ -88,10 +90,15 @@ def forward(  # type:ignore
...
+            raise NotImplementedError(
+                "SliceCombiner does not support multiclass labels."
+            )
+
         predictor_confidences = torch.cat(
             [
-                # Compute the "confidence" using the max score across classes
-                torch.max(F.softmax(output, dim=1), dim=1)[0].unsqueeze(1)
+                # Compute the "confidence" using probability score of the positive class
+                F.softmax(output, dim=1)[:, 0].unsqueeze(-1)
```
[the remark this answers is not in the document]
Comment (vincentschen) [reply snorkel/slicing/modules/slice_combiner.py]: yea, we could. by default, the convention of our binary heads are 2-dimensional, so a softmax over the 2 dims is more natural than a sigmoid over just the first dimension.

ideally, would be nice down the line to have 1-dimensional outputs + a sigmoid—but for consistency, the current convention shouldn't hurt.

File: snorkel/slicing/modules/slice_combiner.py [hunk 5]
```diff
@@ -88,10 +90,15 @@ def forward(  # type:ignore
             flow_dict, self.slice_pred_key
         )
 
+        if predictor_outputs[0].shape[1] > 2:
```
[the remark this answers is not in the document]
Comment (vincentschen) [reply snorkel/slicing/modules/slice_combiner.py]: yes, all `predictor_outputs[i]` would be the same shape

[the remark this answers is not in the document]
Comment (henryre) [reply snorkel/slicing/modules/slice_combiner.py]: no need if they're guaranteed to all be the same

File: snorkel/slicing/modules/slice_combiner.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (henryre) [reply snorkel/slicing/modules/slice_combiner.py]: idk if it's more natural...i agree we should leave as 2-dim so that there's no major overhaul when multiclass is added. however, it's more legible and faster to just do `torch.sigmoid(output[:, 0]).unsqueeze(-1)`
